MapiSmimeToolkit.cpp

```diff
@@ -140,7 +142,13 @@ BOOL ParseArgs(int argc, _TCHAR* argv[], ToolkitOptions * pToolkitOptions)
 				pToolkitOptions->bOverWrite = true;
 				break;
 			case 'd':
-				pToolkitOptions->bDefaultSecurityProfule = true;
+				if (_tcsclen(argv[i]) < 3) {
+					pToolkitOptions->bDefaultSecurityProfule = true;
```

Comment:
Typo: bDefaultSecurityProfule => bDefaultSecurityProfile

Reply:
and really not a fan of just checking length here but I suppose for quick and dirty (very dirty) it kinda works

Reply:
I fixed the typo and made the parsing of the new switch a little more definitive. I don't like it, but it'll do for now.